libraries/AP_AHRS/AP_AHRS.cpp

```diff
@@ -326,3 +326,58 @@ AP_AHRS_View *AP_AHRS::create_view(enum Rotation rotation)
     _view = new AP_AHRS_View(*this, rotation);
     return _view;
 }
+
+/*
+ * Update AOA and SSA estimation based on airspeed, velocity vector and wind vector
+ *
+ * Based on:
+ * "On estimation of wind velocity, angle-of-attack and sideslip angle of small UAVs using standard sensors" by
+ * Tor A. Johansen, Andrea Cristofaro, Kim Sorensen, Jakob M. Hansen, Thor I. Fossen
+ *
+ * "Multi-Stage Fusion Algorithm for Estimation of Aerodynamic Angles in Mini Aerial Vehicle" by
+ * C.Ramprasadh and Hemendra Arya
+ *
+ * "ANGLE OF ATTACK AND SIDESLIP ESTIMATION USING AN INERTIAL REFERENCE PLATFORM" by
+ * JOSEPH E. ZEIS, JR., CAPTAIN, USAF
+ */
+void AP_AHRS::update_AOA_SSA(void)
+{
+    Vector3f aoa_velocity, aoa_wind;
+
+    // get velocity and wind
+    if (get_velocity_NED(aoa_velocity) == false) {
+        return;
+    }
+
+    float vel_len = aoa_velocity.length();
+
+    // do not calculate if speed is too low
+    if (vel_len < 2.0) {
```

Comment:
Sorry, did not quite wake up when answered. But the problem was really present.
The body over air velocity is calculated, that's why wind is needed.
Changed the place of `if (vel_len < 2.0)` and also `vel_len` after that one is calculated.